cmake/HPX_AddModule.cmake

```diff
@@ -62,21 +55,15 @@ function(add_hpx_module libname modulename)
   )
 
   # HPX options
-  if(${_have_compatibility_headers_option})
-    set(_compatibility_headers_default OFF)
-    if(${modulename}_COMPATIBILITY_HEADERS)
-      set(_compatibility_headers_default ON)
-    endif()
-    hpx_option(
-      HPX_${modulename_upper}_WITH_COMPATIBILITY_HEADERS
-      BOOL
-      "Enable compatibility headers for old headers. (default: ${_compatibility_headers_default})"
-      ${_compatibility_headers_default}
-      ADVANCED
-      CATEGORY "Modules"
-      MODULE ${modulename_upper}
-    )
-  endif()
+  hpx_option(
+    HPX_${modulename_upper}_WITH_COMPATIBILITY_HEADERS
+    BOOL
+    "Enable compatibility headers for old headers. (default: ON)"
+    ON
+    ADVANCED
+    CATEGORY "Modules"
+    MODULE ${modulename_upper}
+  )
```

Comment:
I'd go one step further and not have the option at all (just have the headers there until they're removed).

Reply:
But if the warnings are annoying (too much of them) and the users wants to temporarily silence them because of a lack of time to fix them, doesn't it make sense to keep the option still?

Reply:
They still have the option to silence them by defining `HPX_HAVE_DEPRECATION_WARNINGS=0` (and they don't need to rebuild HPX for that to take effect).

Reply:
And more importantly, this option wouldn't have controlled that anyway. That would've been the one removed in #5177.

Reply:
True will remove them